### PR TITLE
Add route, service and proxies for metrics in Console

### PR DESCRIPTION
**Pull request by metalmatze**
This adds the [kube-rbac-proxy](https://github.com/brancz/kube-rbac-proxy) and the [prom-label-proxy](https://github.com/openshift/prom-label-proxy) as sidecars to the prometheus-k8s pod.
The Console will talk a Route that will forward the traffic to a Service that then fowards it through these proxies to Prometheus, making sure that the Console receives only relevant and secured data.

This is an additional ingress to Prometheus for this specific use case.

/cc @brancz @squat

### CODE REVIEW

**Comment by metalmatze** [comment]
I've tested this on my minishift cluster with curl:
```
curl -s -k -H "Authorization: Bearer $TOKEN" 'https://localhost:9092/api/v1/query?query=up&namespace=default' | jq .
```

The `$TOKEN` was base64 decoded from a valid ServiceAccount.

File: jsonnet/prometheus.jsonnet [hunk 1]
```diff
@@ -54,6 +54,27 @@ local namespacesRole = policyRule.new() +
       },
     },
 
+    routeConsole:{
```
Comment (brancz) [inline jsonnet/prometheus.jsonnet]: the openshift console actually proxies the requests into the cluster, so we don't need this route

File: jsonnet/prometheus.jsonnet [hunk 2]
```diff
@@ -82,6 +103,14 @@ local namespacesRole = policyRule.new() +
       service.mixin.spec.withType('ClusterIP') +
       service.mixin.spec.withPorts(servicePort.newNamed('web', 9091, 'web')),
 
+    serviceConsole+:
```
Comment (brancz) [inline jsonnet/prometheus.jsonnet]: I think we should just add an additional `tenancy` port to the existing service

File: jsonnet/prometheus.jsonnet [hunk 3]
```diff
@@ -313,6 +366,32 @@ local namespacesRole = policyRule.new() +
                 },
               ],
             },
+            {
+              name: 'kube-rbac-proxy',
+              image: 'quay.io/brancz/kube-rbac-proxy-dev:v0.3.1-a12535f',
```
Comment (brancz) [inline jsonnet/prometheus.jsonnet]: this needs an update with the now available v0.4.0 release

File: jsonnet/main.jsonnet [hunk 4]
```diff
@@ -20,6 +20,7 @@ local kp = (import 'kube-prometheus/kube-prometheus.libsonnet') +
                  prometheus: 'openshift/prometheus',
                  alertmanager: 'openshift/prometheus-alertmanager',
                  nodeExporter: 'openshift/prometheus-node-exporter',
+                 promLabelProxy: 'quay.io/openshift/prom-label-proxy',
```
Comment (brancz) [inline jsonnet/main.jsonnet]: this needs to be updated

**Comment by metalmatze** [comment]
I addressed your earlier comments and squashed all commits.

File: assets/kube-state-metrics/deployment.yaml [hunk 5]
```diff
@@ -21,7 +21,7 @@ spec:
         - --upstream=http://127.0.0.1:8081/
         - --tls-cert-file=/etc/tls/private/tls.crt
         - --tls-private-key-file=/etc/tls/private/tls.key
-        image: quay.io/coreos/kube-rbac-proxy:v0.3.1
+        image: quay.io/brancz/kube-rbac-proxy:v0.4.0
```
Comment (brancz) [inline assets/kube-state-metrics/deployment.yaml]: pushed images to the coreos repo :slightly_smiling_face:

File: jsonnet/prometheus.jsonnet [hunk 6]
```diff
@@ -313,6 +340,32 @@ local namespacesRole = policyRule.new() +
                 },
               ],
             },
+            {
+              name: 'kube-rbac-proxy',
+              image: $._config.imageRepos.kubeRbacProxy + ':' + $._config.versions.kubeRbacProxy,
+              args: [
+                '--secure-listen-address=0.0.0.0:9092',
+                '--upstream=http://127.0.0.1:9095',
+                '--resource-attributes-file=/etc/kube-rbac-proxy/resource-attributes.yaml',
```
Comment (brancz) [inline jsonnet/prometheus.jsonnet]: this flag is renamed in v0.4.0 to `--config-file`

File: jsonnet/prometheus.jsonnet [hunk 7]
```diff
@@ -156,6 +159,29 @@ local namespacesRole = policyRule.new() +
       secret.mixin.metadata.withNamespace($._config.namespace) +
       secret.mixin.metadata.withLabels({ 'k8s-app': 'prometheus-k8s' }),
 
+    kubeRbacProxySecret:
+      local config = {
+        'resource-attributes.yaml': std.base64(std.manifestYamlDoc({
```
Comment (brancz) [inline jsonnet/prometheus.jsonnet]: as this now contains more than resource attribute configuration, let's rename this to `config.yaml`

**Comment by metalmatze** [comment]
@brancz Fixed the wrong image repositories and updated the config to be called `config.yaml` now.

**Comment by brancz** [comment]
@metalmatze looks like there was already another merge that caused a conflict :S

**Comment by brancz** [comment]
if merged in time, maybe we can include https://github.com/coreos/prometheus-operator/pull/2055 in this PR?

**Comment by metalmatze** [comment]
It was a conflict in `bindata.go`. :grin: 
I've also updated the kube-prometheus dependency to the latest commit on its master branch.

**Comment by brancz** [comment]
Nice job! :+1: 

/lgtm

**Pull request merged**